### PR TITLE
Translate action

### PR DESCRIPTION
**Pull request by balteravishay**
This PR adds an action that translate a document to a list of languages.

### CODE REVIEW

File: .github/workflows/translation.yml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          set -eux
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sas=`az storage container generate-sas -n $INPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions dlrw --expiry $end -o tsv`
+          az storage blob upload \
+            --overwrite true \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --container-name $INPUT_CONTAINER_NAME \
+            --name ${{ vars.DOCUMENT_PATH }} \
+            --file ${{ vars.DOCUMENT_PATH }} \
+            --sas-token $sas
```
Comment (SharonHart) [inline .github/workflows/translation.yml:93]: Can be a reusable workflow for upload x to y

File: .github/workflows/translation.yml [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sas=`az storage container generate-sas -n $GLOSSARY_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions dlrw --expiry $end -o tsv`
+          az storage blob upload \
+            --overwrite true \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --container-name $GLOSSARY_CONTAINER_NAME \
+            --name ${{ matrix.glossary }} \
+            --file translations/${{ matrix.language-id }}/${{ matrix.glossary }} \
+            --sas-token $sas
+          glossary_sas=`az storage container generate-sas -n $GLOSSARY_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lre --expiry $end -o tsv`
```
Comment (SharonHart) [inline .github/workflows/translation.yml:153]: isn't it $sas?

File: .github/workflows/translation.yml [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+          glossary_uri_with_sas=https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$GLOSSARY_CONTAINER_NAME/${{ matrix.glossary }}?$glossary_sas
```
Comment (SharonHart) [inline .github/workflows/translation.yml:154]: can't be fetched from the storage blob upload command?

File: .github/workflows/translation.yml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+          glossary_json=",\"glossaries\": [{\"glossaryUrl\":\"$glossary_uri_with_sas\",\"format\":\"CSV\"}]"
+          echo "glossary_json=$glossary_json" >> $GITHUB_ENV
+      - name: "Upload Translation"
+        if: ${{ ! matrix.glossary }}
+        run: |
+          set -eux
+          echo "glossary_json= " >> $GITHUB_ENV
```
Comment (SharonHart) [inline .github/workflows/translation.yml:161]: twice?

File: .github/workflows/translation.yml [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        run: |
+          set -ux
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          input_sas=`az storage container generate-sas -n $INPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lr --expiry $end -o tsv`
+          output_sas=`az storage container generate-sas -n $OUTPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lw --expiry $end -o tsv`
+
+          operation_id=$(az rest --method post \
+            --url ${{ secrets.TRANSLATOR_API_URI }}translator/text/batch/v1.0/batches \
+            --headers Ocp-Apim-Subscription-Key=${{ secrets.TRANSLATOR_API_KEY }} \
+            --body "{\"inputs\": [{\"source\": { \"sourceUrl\": \"https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$INPUT_CONTAINER_NAME/?$input_sas\"}, \"targets\": [{\"targetUrl\": \"https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$OUTPUT_CONTAINER_NAME?$output_sas\",\"language\": \"${{ matrix.language-id }}\" $glossary_json }]}]}" \
```
Comment (SharonHart) [inline .github/workflows/translation.yml:176]: maybe from a json checked into the repo with placeholders?

File: .github/workflows/translation.yml [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+      ############################
+      # Wait for Translation     #
+      ############################
+      - name: "Wait for translation job"
+        run: |
+          set -eux
+          action_state="${{ env.action_state }}"
+          echo $action_state
+          status=NotStarted
+          while [ "$status" = "NotStarted" ]
```
Comment (SharonHart) [inline .github/workflows/translation.yml:189]: "$status" = "NotStarted" or "$status" = "Running"

Comment (SharonHart) [reply .github/workflows/translation.yml:189]: or != Succeeded

File: .github/workflows/translation.yml [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+          az storage blob download \
+          --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+          --container-name $OUTPUT_CONTAINER_NAME \
+          --name  ${{ vars.DOCUMENT_PATH }} \
+          --file  ./translations/${{ matrix.language-id }}/${{ vars.DOCUMENT_PATH }} \
+          --sas-token $sas
+      #############################
+      # Delete output container   #
+      #############################
+      - name: Delete target container
```
Comment (SharonHart) [inline .github/workflows/translation.yml:229]: runs on failure?

File: .github/workflows/translation.yml [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+          set -eux
+          az storage container delete \
+            --name $OUTPUT_CONTAINER_NAME \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --auth-mode login
+      #############################
+      # Create PR for language    #
+      #############################
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
```
Comment (SharonHart) [inline .github/workflows/translation.yml:240]: nice

File: .github/workflows/translation.yml [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+        with:
+          branch: feature/${{ matrix.language }}-translation
```
Comment (SharonHart) [inline .github/workflows/translation.yml:242]: when do you push to the branch?

File: .github/workflows/translation.yml (hunk 1 above)
Comment (balteravishay) [reply .github/workflows/translation.yml]: fixed so there is no need to

File: .github/workflows/translation.yml (hunk 2 above)
Comment (balteravishay) [reply .github/workflows/translation.yml]: it wasn't (different permissions), but found a way to not need it anyway :)

File: .github/workflows/translation.yml (hunk 3 above)
Comment (balteravishay) [reply .github/workflows/translation.yml]: no, it has sas

File: .github/workflows/translation.yml (hunk 5 above)
Comment (balteravishay) [reply .github/workflows/translation.yml]: done, good one

File: .github/workflows/translation.yml (hunk 6 above)
Comment (balteravishay) [reply .github/workflows/translation.yml]: done, good one

File: .github/workflows/translation.yml (hunk 7 above)
Comment (balteravishay) [reply .github/workflows/translation.yml:229]: fixed

File: .github/workflows/translation.yml (hunk 9 above)
Comment (balteravishay) [reply .github/workflows/translation.yml:243]: the x-translation branch is updated if it wasn't pushed to main.

**Pull request merged**